src/fluree/db/indexer/default.cljc

```diff
@@ -292,10 +291,10 @@
              (let [child (peek stack*)]
                (if (and child
                         (index/descendant? node child))     ; all of a resolved
-                 ; branch's children
-                 ; should be at the top
-                 ; of the stack
-                 (recur (conj child-nodes child)
+                                                            ; branch's children
+                                                            ; should be at the top
+                                                            ; of the stack
+                 (recur (conj child-nodes (index/unresolve child))
```

Comment:
I assume this is to allow for GC... if so, could this be part of the very large ledger reindexing slow-downs we saw in v2?

Reply:
Yes, and yes. Without this, the whole index tree sits on the database during a reindex.